src/Idris/AbsSyntax.hs

```diff
@@ -390,7 +379,7 @@ addNameHint ty n
         putIState $ i { idris_namehints = addDef ty' ns' (idris_namehints i) }
 
 getNameHints :: IState -> Name -> [Name]
-getNameHints i (UN arr) | arr == txt "->" = [sUN "f",sUN "g"]
+getNameHints _ (UN arr) | arr == txt "->" = [sUN "f",sUN "g"]
```

Comment:
I usually prefer named ignored patterns, e.g. `_i`. It's not that important but I think it's something to consider.